packages/runtime/container-runtime/src/summaryManagerUtils.ts

```diff
@@ -0,0 +1,200 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IEvent } from "@fluidframework/common-definitions";
+import { assert, TypedEventEmitter } from "@fluidframework/common-utils";
+import { IQuorum, ISequencedClient } from "@fluidframework/protocol-definitions";
+
+export const summarizerClientType = "summarizer";
+
+export interface ITrackedClient {
+    readonly clientId: string;
+    readonly sequenceNumber: number;
+}
+
+interface ILinkedClient extends ITrackedClient {
+    prevClient: ILinkedClient | undefined;
+    nextClient: ILinkedClient | undefined;
+}
+
+interface IFullClient extends ILinkedClient {
+    readonly isSummarizer: boolean;
+}
+
+interface IOrderedClientsEvents extends IEvent {
+    (event: "currentChange", listener: (client: ITrackedClient | undefined) => void);
+    (event: "summarizerChange", listener: (summarizerCount: number) => void);
+}
+
+export class OrderedClients extends TypedEventEmitter<IOrderedClientsEvents> {
+    private readonly clientMap = new Map<string, IFullClient>();
+    private readonly rootClient: ILinkedClient = {
+        sequenceNumber: -1,
+        clientId: "",
+        prevClient: undefined,
+        nextClient: undefined,
+    };
+    private lastClient: ILinkedClient = this.rootClient;
```

Comment:
it's a bit hard to reason over all the relavite terms used here, as it's not always immediatly clear if you are refereing to the list, or the clients, also not clear without walking all the logic on what the order of the list is

Reply:
Renamed to oldestClient